test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,66 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/disruption"
+	"github.com/openshift/origin/test/extended/util/disruption/controlplane"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+func forceApiRollout(oc *exutil.CLI) {
+	redeploymentReason := fmt.Sprintf(`{"spec":{"forceRedeploymentReason":"resiliency-test-%v"}}`, uuid.NewUUID())
+
+	_, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType,
+		[]byte(redeploymentReason), metav1.PatchOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+func isApiUp(client clientset.Interface) (bool, error) {
+	_, err := client.CoreV1().RESTClient().Get().RequestURI("/readyz").DoRaw(context.TODO())
+	return err == nil, err
+}
+
+var _ = ginkgo.Describe("[Conformance][sig-api-machinery][Serial]", func() {
+	f := framework.NewDefaultFramework("cluster-resiliency")
+	f.SkipNamespaceCreation = true
+	f.SkipPrivilegedPSPBinding = true
+
+	oc := exutil.NewCLIWithoutNamespace("cluster-resiliency")
+
+	ginkgo.It("should allow a fast kube-apiserver rollout", func() {
```

Comment:
if you want to add operators here as well, rename it (= remove mentioning kube-apiserver and mention the tested operators). In the way it is named now, API team will get bugs due to operators. That's wrong.